pkg/db/task.go

```diff
@@ -0,0 +1,228 @@
+	tasks := make([]*Task, 0, limit)
+	db := GetDB()
+	time, err := time.Parse("02.01.2006", search)
+	if err == nil {
+		query := `
+			SELECT * FROM scheduler WHERE date = ? LIMIT ?
+			`
+		rows, err := db.Query(query, time.Format(DateFormat), limit)
+		if err != nil {
+			return tasks, err
+		}
+		defer rows.Close()
+
+		for rows.Next() {
+			var task Task
+
+			err := rows.Scan(&task.Id, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+			if err != nil {
+				return tasks, fmt.Errorf("ошибка сканирования строки: %w", err)
+			}
+			tasks = append(tasks, &task)
+
+		}
+
+		if err := rows.Err(); err != nil {
+			return nil, fmt.Errorf("ошибка итерации по строкам: %w", err)
+		}
+
+		return tasks, nil
+	} else {
+		query := `
+			SELECT id, date, title, comment, repeat
+			FROM scheduler
+			WHERE LOWER(title) LIKE LOWER(?) COLLATE NOCASE OR LOWER(comment) LIKE LOWER(?) COLLATE NOCASE
+			ORDER BY date
+			LIMIT ?
+		`
+
+		db := GetDB()
+
+		search = strings.ToLower(search)
+		likePattern := "%" + search + "%"
+		// likePattern := search
+
+		log.Printf("Исходный search: '%s'", search)
+		log.Printf("Сформированный likePattern: '%s'", likePattern)
+		log.Printf("Выполняемый запрос: %s", query)
+		log.Printf("Параметры запроса: pattern1=%s, pattern2=%s, limit=%d", likePattern, likePattern, limit)
+
+		rows, err := db.Query(query, likePattern, likePattern, limit)
+		// rows, err := db.Query(query, likePattern, limit)
+
+		if err != nil {
+			return nil, fmt.Errorf("ошибка выполнения запроса: %w (query: %s, params: %v, %v, %v)",
+				err, query, likePattern, likePattern, limit)
+		}
+		defer rows.Close()
+
+		var found bool
+
+		for rows.Next() {
+			var task Task
+			err := rows.Scan(&task.Id, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+			if err != nil {
+				return nil, fmt.Errorf("ошибка сканирования строки: %w", err)
+			}
+			tasks = append(tasks, &task)
+			found = true
+			log.Printf("Найдена запись: ID=%s, Title='%s'", task.Id, task.Title)
+		}
+
+		if !found {
+			log.Printf("Записи не найдены для search='%s'", search)
+		}
+
+		if err := rows.Err(); err != nil {
+			return nil, fmt.Errorf("ошибка итерации по строкам: %w", err)
+		}
+
+		return tasks, nil
+	}
+}
+
+func GetTask(id int) (Task, error) {
+	db := GetDB()
+	var task Task
+
+	query := `
+        SELECT id, date, title, comment, repeat 
+        FROM scheduler 
+        WHERE id = ?
+    `
+	err := db.QueryRow(query, id).Scan(&task.Id, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+	if err != nil {
+		return task, err
+	}
+
+	return task, nil
+}
+
+func EditTask(task *Task) error {
+	db := GetDB()
+
+	query := `
+        UPDATE scheduler 
+        SET title = ?, 
+            comment = ?, 
+            date = ?, 
+            repeat = ? 
+        WHERE id = ?
+    `
+
+	result, err := db.Exec(query, task.Title, task.Comment, task.Date, task.Repeat, task.Id)
+
+	if err != nil {
+		return fmt.Errorf("некорректное обновление задачи: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
		return fmt.Errorf("некорректное обновление задачи: %w", err)
```